### ISSUE TITLE
v6.1.7 introduces NPE if component is created without config

### ISSUE THREAD

**Issue by gkamperis** [ngx-formly/ngx-formly#3700]
Added comment on the PR. 

In a default test that just gets the custom component to be created without a field we get a NPE. 

https://github.com/ngx-formly/ngx-formly/commit/7c90ae4141a91759dc2b1e664b8c7b0e0467a2b8#r114930691

Same happens on v6.1.6 for
https://github.com/ngx-formly/ngx-formly/blob/e1881e5f0ab2c4aaa54a49c90c179978b7ba45be/src/core/src/lib/templates/field.type.ts#L49

To reproduce you only need to 

```
beforeEach(() => {
    fixture = TestBed.createComponent(CustomFormlyTypeComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

it('should create', () => {
    expect(component).toBeTruthy();
});
```

essentially just create the component without giving it a `field`

**Comment by gkamperis** [comment]
@aitboudad any looks into this? this is quite an issue that has been introduced as the `field` is not always available anymore so its typedef is incorrect

**Comment by gkamperis** [comment]
@aitboudad here is a repro from within the git repo itself

```
describe('FormlyField Component', () => {
  fit('should create', async () => {
    await TestBed.configureTestingModule({
      declarations: [FormlyOnPushComponent],
      imports: [ReactiveFormsModule, FormlyModule.forRoot()],
      teardown: { destroyAfterEach: false },
    }).compileComponents();

    const fixture = TestBed.createComponent(FormlyOnPushComponent);
    fixture.detectChanges();
    expect(fixture).toBeTruthy();
  });
```

![image](https://github.com/ngx-formly/ngx-formly/assets/18055519/517441b7-be75-4801-b84e-520dad141956)

**Comment by aitboudad** [comment]
Please note: for testing a custom type check https://github.com/ngx-formly/ngx-formly/issues/2723#issue-818214519

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.1.8](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.1.8) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
